src/exercises/Exercises.js

```diff
@@ -45,12 +62,88 @@ export default function Exercises({ api, articleID }) {
   }, []);
 
   function initializeExercises(bookmarks, title) {
-    setbookmarksToStudyList(bookmarks);
     NUMBER_OF_EXERCISES = bookmarks.length;
-    setCurrentBookmarkToStudy(bookmarks[currentIndex]);
+    setCurrentExerciseSession(bookmarks);
     setTitle(title);
   }
 
+  function setCurrentExerciseSession(bookmarks) {
+    let bookmarkSum = EXERCISES.reduce((a, b) => a + b.requiredBookmarks, 0);
+    let batches = parseInt(NUMBER_OF_EXERCISES / bookmarkSum);
```

Comment:
can you pls. add a comment for what batches is? I'm not sure I get it.

ah, it's possibleBatches? 
and NUMBER_OF_EXERCISES is actually BOOKMARKS_TO_PRACTICE? 

ok. I'll try some "refactoring for understanding" in d6ccdba6f722ad6d7d80878ecb01c88f35a927f4